src/commands/rated_self_report.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+from pathlib import Path
+
+import click
+import requests
+from eth_typing import HexAddress
+
+from src.common.graph import GraphClient
+from src.common.validators import validate_eth_address
+from src.common.vault_config import VaultConfig
+from src.config.settings import AVAILABLE_NETWORKS, DEFAULT_NETWORK, settings
+
+
+@click.option(
+    '--data-dir',
+    default=str(Path.home() / '.stakewise'),
+    envvar='DATA_DIR',
+    help='Path where the vault data will be placed. Default is ~/.stakewise.',
+    type=click.Path(exists=True, file_okay=False, dir_okay=True),
+)
+@click.option(
+    '--vault',
+    prompt='Enter your vault address',
+    help='The vault address.',
+    type=str,
+    callback=validate_eth_address,
+)
+@click.option(
+    '--network',
+    default=DEFAULT_NETWORK,
+    help='The network of your vault.',
+    prompt='Enter the network name',
+    type=click.Choice(
+        AVAILABLE_NETWORKS,
+        case_sensitive=False,
+    ),
+)
+@click.option(
+    '--pool-tag',
+    default='',
+    help='The pool name listed on the Explorer.',
+    prompt='Enter the pool tag (if any)',
+    type=str,
+)
+@click.option(
+    '--token',
+    prompt='Enter your OAuth token',
+    help='OAuth token for authorization.',
+    type=str,
+)
+@click.command(help='Self-report your validators to the Rated Network.')
+def rated_self_report(
+    vault: HexAddress,
+    network: str,
+    pool_tag: str,
+    token: str,
+    data_dir: str,
+) -> None:
+    if network == 'gnosis':
```

Comment:
better to check if networks in supported networks. That about chiado foe example